api/src/services/lottery.service.ts

```diff
@@ -224,4 +257,325 @@ export class LotteryService {
+                : false,
+              applicationLotteryPositions: forLottery
+                ? {
+                    select: {
+                      ordinal: true,
+                    },
+                    where: {
+                      multiselectQuestionId: null,
+                    },
+                  }
+                : false,
+            },
+            where: {
+              listingId: queryParams.listingId,
+              deletedAt: null,
+              markedAsDuplicate: forLottery ? false : undefined,
+              id: {
+                in: applications
+                  .slice(i, i + NUMBER_TO_PAGINATE_BY)
+                  .map((app) => app.id),
+              },
+            },
+          });
+          if (forLottery) {
+            paginatedApplications = paginatedApplications.sort(
+              (a, b) =>
+                a.applicationLotteryPositions[0].ordinal -
+                b.applicationLotteryPositions[0].ordinal,
+            );
+          }
+          const rows: Partial<Row>[] = [];
+          paginatedApplications.forEach((app) => {
+            const row: Partial<Row> = {};
+            let preferences: ApplicationMultiselectQuestion[];
+            let programs: ApplicationMultiselectQuestion[];
+            csvHeaders.forEach((header) => {
+              let multiselectQuestionValue = false;
+              let parsePreference = false;
+              let parseProgram = false;
+              let value = header.path.split('.').reduce((acc, curr) => {
+                // return preference/program as value for the format function to accept
+                if (multiselectQuestionValue) {
+                  return acc;
+                }
+
+                if (parsePreference) {
+                  // curr should equal the preference id we're pulling from
+                  if (!preferences) {
+                    preferences =
+                      app.preferences as unknown as ApplicationMultiselectQuestion[];
+                  }
+                  parsePreference = false;
+                  // there aren't typically many preferences, but if there, then a object map should be created and used
+                  const preference = preferences.find(
+                    (preference) => preference.key === curr,
+                  );
+                  multiselectQuestionValue = true;
+                  return preference;
+                } else if (parseProgram) {
+                  // curr should equal the preference id we're pulling from
+                  if (!programs) {
+                    programs =
+                      app.programs as unknown as ApplicationMultiselectQuestion[];
+                  }
+                  parsePreference = false;
+                  // there aren't typically many programs, but if there, then a object map should be created and used
+                  const program = programs.find(
+                    (preference) => preference.key === curr,
+                  );
```

Comment:
Just for safety reasons we should port over the null check that was added here https://github.com/metrotranscom/doorway/pull/768